### PR TITLE
[fix][bluray] Fix Blu-Ray playback for all playback paths and refactor ShowPlaySelection

### PR DESCRIPTION
**Pull request by ace20022**
This fixes an issue first described in http://forum.kodi.tv/showthread.php?tid=214883.
There are so many paths to start playback of a file, that I move the selection dialog call to the confluence of these paths.

### CODE REVIEW

**Comment by ace20022** [comment]
Maybe @topfs2 @Montellese for review? 

Should I better remove the ```CFileItemPtr``` "hack" and rewrite ShowPlaySelection instead?

**Comment by ace20022** [comment]
And ping @Paxxi ;)

**Comment by ace20022** [comment]
@mkortstiege thanks a lot!

jenkins build this please

File: xbmc/dialogs/GUIDialogSimpleMenu.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "FileItem.h"
+
```
Comment (ace20022) [inline xbmc/dialogs/GUIDialogSimpleMenu.h]: Why?  All builds are green.

Comment (FernetMenta) [inline xbmc/dialogs/GUIDialogSimpleMenu.h]: it only works by chance because string is already included by the files using this.

Comment (ace20022) [inline xbmc/dialogs/GUIDialogSimpleMenu.h]: Shame on me. Like a famous German said: "again what learned" ;)

File: xbmc/dialogs/GUIDialogSimpleMenu.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  CFileItemList items;
+
+  if (!XFILE::CDirectory::GetDirectory(directory, items, XFILE::CDirectory::CHints(), true))
+  {
+    CLog::Log(LOGERROR, "CGUIWindowVideoBase::ShowPlaySelection - Failed to get play directory for %s", directory.c_str());
+    return true;
+  }
+
+  if (items.Size() == 0)
```
Comment (fritsch) [inline xbmc/dialogs/GUIDialogSimpleMenu.cpp]:
```
int CFileItemList::Size() const
{
  CSingleLock lock(m_lock);
  return (int)m_items.size();
}

bool CFileItemList::IsEmpty() const
{
  CSingleLock lock(m_lock);
  return (m_items.size() <= 0);
}
```

Hehe, wanted to bikeshed about the Size(), but then saw the implementation :-) both is equally bad.

Comment (ace20022) [inline xbmc/dialogs/GUIDialogSimpleMenu.cpp]: jup strange. (that part is just moved btw)

Comment (fritsch) [inline xbmc/dialogs/GUIDialogSimpleMenu.cpp]: As that m_items is a typedef of a vector, both size() and empty() have the very same computing time, both are constant, so it does not matter that much, only when we start to change the underlaying vector.

Comment (ace20022) [inline xbmc/dialogs/GUIDialogSimpleMenu.cpp]: But it's still strange to cast to int and to check an unsigned for \leq 0

**Comment by ace20022** [comment]
rewrote it. Now it uses fileitem ref instead of shared pointers. I also remove thequalifier from Play* methods in application. Any thoughts?

**Comment by Paxxi** [comment]
Merge the last two commits maybe since they're related

**Comment by ace20022** [comment]
> Squash the last two commits maybe since they're related

Have you seen the commit message ;)
If no objections I'll merge it tomorrow.

**Comment by Paxxi** [comment]
I read it as you tried to squash but missed a commit :) no objections here

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
check for ignore list membership...

jenkins build this please

**Comment by ace20022** [comment]
jenkins build this please

**Pull request merged**